src/browser/ui/ReactDOMComponent.js

```diff
@@ -234,8 +247,13 @@ ReactDOMComponent.Mixin = {
           }
           propValue = CSSPropertyOperations.createMarkupForStyles(propValue);
         }
-        var markup =
-          DOMPropertyOperations.createMarkupForProperty(propKey, propValue);
+        var markup = null;
+        if (this._tag != null && this._tag.indexOf('-') >= 0) {
+          validateDangerousAttributeName(propKey);
+          markup = DOMPropertyOperations.createMarkupForCustomAttribute(propKey, propValue);
```

Comment:
Or @spicyj if we don't feel strongly about it, we can keep it as-is (as per reasons above) and we can always change it later.